cpp/include/cudf/utilities/span.hpp

```diff
@@ -278,7 +278,11 @@ struct host_span : public cudf::detail::span_base<T, Extent, host_span<T, Extent
    * @param idx the index of the element to access
    * @return A reference to the idx-th element of the sequence, i.e., `data()[idx]`
    */
-  constexpr typename base::reference operator[](size_type idx) const { return this->_data[idx]; }
+  constexpr typename base::reference operator[](typename base::size_type idx) const
+  {
+    static_assert(sizeof(idx) >= sizeof(size_t), "index type must not be smaller than size_t");
+    return this->_data[idx];
+  }
```

Comment:
I don't think the `static_assert` statement works as expected. If we pass in an `int` value, it can be implicitly promoted into `size_t` and this will not complain at all.

Reply:
A better solution to this IMO is to use function template:
```
template<typename T>
operator[](T idx) const
{
  static_assert(sizeof(T) >= sizeof(size_t));
}
```